### DISCUSSION TITLE
Bind: Cannot assign requested address

### DISCUSSION THREAD

**Discussion by Hammaad-M** [pocketbase/pocketbase#3938] [Q&A] [answered]
I've tried everything shared from this [thread ](https://github.com/pocketbase/pocketbase/issues/107) to setup pocketbase on an Azure VM, and still get `bind: cannot assign requested address`. First, I tried just running the executable `./pocketbase serve --http="<public_ip>:80"` and received the error above. I then setup a reverse proxy with Nginx to listen at port 80 and forward all requests to `127.0.0.1:8090` where I ran the executable--but can't access the admin dashboard at `<public_ip>:80/_`.

here is my Nginx config: 
`
server {
        listen 80;
        location / {
                proxy_pass http://127.0.0.1:8090;
        }
}
`
running `./pocketbase serve --http="127.0.0.1:8090"` works, but I can't access the admin portal at `<public_ip>:80/_`.

I then got the private IP using `hostname -I`, which is `10.0.0.4`. I added the following line to etc/hosts: `10.0.0.4 <public_ip>`. Tried to run executable at <public_ip>:80, swapping 80 with every port I know, but get `bind: cannot assign requested address`. Then tried running at `10.0.0.4:80`, which ran successfully but admin dashboard is not accessible at the public ip. 

Essentially, I can't run the executable on the public IP, but can run it on the local IP--only then I can't access the admin dashboard from the browser. Using Nginx to proxy and the hosts file is simply not working. I'm all out of ideas. I even tried starting over on a google VM instance, but get the same error.

I don't want to get a domain for my database, since I can use the IP when querying it from the frontend. 

Any help would be much appreciated...

**Comment by ganigeorgiev** [comment, marked as answer]
The description of the issue is very vague and it is difficult to say what exactly is the cause of it. You didn't provide the exact steps or commands you've performed and there isn't really enough details to help you debug it. I also don't have much experience with Azure and don't know how did you configure the vm nor how do you even access it.

If you are asking for a minimal deployment guide, please follow https://pocketbase.io/docs/going-to-production/#minimal-setup.

And again, if you get `bind: cannot assign requested address` then this could mean that there is either another application already running on the same address or your user maybe doesn't have permission to bind to your port. If you have installed nginx and you are not going to use it, try to stop it or unninstall it entirely to prevent keeping the 80 port.

My recommendation is:
- start from a clean vm state
- deploy the binary on your vm
- and since you want to use PocketBase with IP:80 address, start it with:
  ```sh
  # 0.0.0.0 means bind to all IPv4 interfaces
  ./pocketbase serve --http="0.0.0.0:80"
  ```

Then find the public IPv4 address of your vm and try to access it in the browser with `http://YOUR_IP/_/`.

**Comment by Hammaad-M** [comment]
Okay so I got it working finally. When setting up a fresh VM, I found that the 80 and 443 ports are disabled by default on Azure VMs. I enabled them, then ran the following cmds:
`
chmod 777 ./ # open all restrictions on current directory, used FileZilla to upload pocketbase binary
sudo setcap 'cap_net_bind_service=+ep' ./pocketbase # give pocketbase permissions
sudo chmod 777 ./pocketbase # get permission to run
./pocketbase serve http="0.0.0.0:80"
`
I can now access the admin dashboard at <public_ip>:80! 

I also tried running `./pocketbase serve https="0.0.0.0:443`, which runs but when I try to access `https://<public_ip>:443`, I get "Connection not secure in the browser", and the following error in the terminal: `http: TLS handshake error from <some_ip>:64854: acme/autocert: missing server name` 

Thank you @ganigeorgiev for your help, and let me know if there is something I'm missing for the https error.

**Reply by ganigeorgiev** [reply]
For the https error see https://github.com/pocketbase/pocketbase/discussions/452#discussioncomment-3650417.

In short - it is not possible to issue a Let's encrypt certificate if you don't have a public domain name.
